packages/workspace/src/utils/buildale-libs-utils.ts

```diff
@@ -62,6 +63,31 @@ export function calculateProjectDependencies(
   return { target, dependencies };
 }
 
+interface NgPackagrOptions {
+  dest?: string;
+  [key: string]: any;
+}
+
+export function getNgPackgrOptions(
+  basePath: string
+): NgPackagrOptions | undefined {
```

Comment:
Thanks for working on this!

It looks like this function will suffer the same issue that you are solving - locations to `ng-package.json` here are hardcoded based on some assumptions.

Those may very well be the default lookup locations but first thing to look for is the build target of the project library in `angular.json` because that's usually where the path to `ng-package.json` is specified by the user.

Also it probably make sense to do this specific config lookup only when the builder specified is `@angular-devkit/build-ng-packagr:build` - since, I assume, in other builders configuration might be different...

Example:
```jsonc
"architect": {
  "build": {
    "builder": "@angular-devkit/build-ng-packagr:build",
    "options": {
      "tsConfig": "libs/my-lib/tsconfig.lib.json",
      "project": "libs/my-lib/oops/ng-package.json" // <-- Pick this path first
    }
  }
}
```